### PR TITLE
Add RBS signatures for customizations of s3

### PR DESCRIPTION
**Pull request by ksss**
I wrote the RBS for the customizations code.

When performing type checks, emphasis is placed on correcting undefined method errors, and the types of arguments are not specified in detail.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

1. To make sure we include your contribution in the release notes, please make sure to add description entry for your changes in the "unreleased changes" section of the `CHANGELOG.md` file (at corresponding gem). For the description entry, please make sure it lives in one line and starts with `Feature` or `Issue` in the correct format.

2. For generated code changes, please checkout below instructions first:
  https://github.com/aws/aws-sdk-ruby/blob/version-3/CONTRIBUTING.md

Thank you for your contribution!

### CODE REVIEW

File: gems/aws-sdk-s3/spec/object_summary_spec.rb [hunk 1]
```diff
@@ -52,31 +52,31 @@ module S3
 
       it 'delegates #presigned_url to #object' do
         expect(object).to receive(:presigned_url)
-          .with('method', { opt_name: 'opt-value' })
-        object_summary.presigned_url('method', opt_name: 'opt-value')
+          .with(:get, { opt_name: 'opt-value' }).and_return('return-value')
```
Comment (ksss) [inline gems/aws-sdk-s3/spec/object_summary_spec.rb]: `rake rbs:spytest` was failing and needed to be fixed.

File: gems/aws-sdk-s3/sig/customizations/object.rbs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      def move_to: (untyped target, ?Hash[Symbol, untyped] options) -> void
+
+      # https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Object.html#presigned_post-instance_method
+      def presigned_post: (?Hash[Symbol, untyped]) -> untyped
+
+      # https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Object.html#presigned_url-instance_method
+      def presigned_url: ((:get | :put | :head | :delete | :create_multipart_upload | :list_multipart_uploads | :complete_multipart_upload | :abort_multipart_upload | :list_parts | :upload_part) method, ?Hash[Symbol, untyped] params) -> String
+
+      # https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Object.html#presigned_request-instance_method
+      def presigned_request: ((:get | :put | :head | :delete | :create_multipart_upload | :list_multipart_uploads | :complete_multipart_upload | :abort_multipart_upload | :list_parts | :upload_part) method, ?Hash[Symbol, untyped] params) -> [String, Hash[String, String]]
```
Comment (mullermp) [inline gems/aws-sdk-s3/sig/customizations/object.rbs]: I would have to check (on mobile right now) but I think the presigned methods can take any operation as a symbol. Does it make sense to be less exhaustive here?

**Review by mullermp — COMMENTED** [review]
Thanks for adding these! At a glance these look ok. I can take a deeper look tomorrow night / Monday and get it merged then. Approving CI for you now.

File: gems/aws-sdk-s3/sig/customizations/object.rbs (hunk 2 above)
Comment (ksss) [reply gems/aws-sdk-s3/sig/customizations/object.rbs]: I followed the documentation and did not know that other values could be valid.
If you want to be more lax, we can use `String | Symbol` instead.

Comment (mullermp) [reply gems/aws-sdk-s3/sig/customizations/object.rbs]: Up to you, I think documenting Symbol makes sense. It can be any method.

**Review by mullermp — APPROVED** [review]
Thank you! If you can address that one change (or provide input otherwise), I'd be happy to merge this.

File: gems/aws-sdk-s3/sig/customizations/object.rbs (hunk 2 above)
Comment (ksss) [reply gems/aws-sdk-s3/sig/customizations/object.rbs]: The purpose of this PR is to reduce the number of undefined methods. Argument types should be loosely specified.

**Comment by ksss** [comment]
Thank you for reviewing. I resolved!

**Review by alextwoods — APPROVED** [review]
Thanks for adding these!

**Pull request merged**